main/lsp/requests/hover.cc

```diff
@@ -14,22 +14,35 @@ namespace sorbet::realmain::lsp {
 string methodInfoString(const core::GlobalState &gs, const core::TypePtr &retType,
                         const core::DispatchResult &dispatchResult,
                         const unique_ptr<core::TypeConstraint> &constraint) {
-    string contents;
+    vector<string> contents;
     auto start = &dispatchResult;
-    ;
+
     while (start != nullptr) {
         auto &component = start->main;
         if (component.method.exists()) {
-            if (!contents.empty()) {
-                contents += "\n";
+            auto method = component.method;
+
+            auto localRetType = start->returnType;
+            auto &constraint = component.constr;
+            if (constraint) {
+                localRetType = core::Types::instantiate(gs, localRetType, *constraint);
             }
-            contents = absl::StrCat(
-                contents, prettyTypeForMethod(gs, component.method, component.receiver, retType, constraint.get()));
+
+            contents.push_back(
+                prettySigForMethod(gs, method, component.receiver, localRetType, constraint.get()));
```

Comment:
I've added this as a commit I intend on reverting, but helps to illustrate my point better. 

With aiming for the hover of the following, I've come across some issues and would love some feedback/insight:
```
sig {params(x: Integer).returns(String)}
sig {params(y: T.any(Integer, String)).returns(Integer)}

-> T.any(Integer, String)
```

My first approach was to pass `component.method.data(gs)->resultType` here, but this broke other parts of the code that were relying on instantiation of the type. A couple of examples of that being:
```
sig {params(a: Integer).returns(T.type_parameter(:A))}
sig {params(arg0: Integer, arg1: String).returns(Proc2::Return)}
```

Switching to the above mitigates that for those cases (the same as using `retType` passed in). However in the union case we will get the following result: 
```
sig {returns(T.any(String, Integer))}
sig {returns(Integer)}
```
This occurs because `Types::Instantiate` will create the fully realized type including all of the nested methods. 

Is there an existing way to solve for both of these cases? Ideally we would have generics, procs, etc be realized, but that nested types aren't considered as a part of that. It feels like we need to be able to solve for both since you could have a generic as a part of a union, in which case you want to print its type, then it's nested type in the `-> <type>` portion.